application/controllers/Lotw.php

```diff
@@ -525,7 +525,10 @@ private function loadFromFile($filepath, $display_view = "TRUE")
 					}
 					// Present only if the QSLing station specified a single valid grid square value in its station location uploaded to LoTW.
 					if (isset($record['gridsquare'])) {
-						$qsl_gridsquare = $record['gridsquare'];
+						// Only overwrite if incoming grid is at least of the same length/precision than existing in the QSO record
+						if (strlen($record['gridsquare']) >= strlen($status[2])) {
```

Comment:
here is missing something. if the $record['gridsquare'] is set but has not the same precision as the origin record it does not define $qsl_gridsquare. I would suggest add another else $qsl_gridsquare = ""; Thx @int2001